model/websocket_client.go

```diff
@@ -60,7 +60,10 @@ func NewWebSocketClientWithDialer(dialer *websocket.Dialer, url, authToken strin
 
 	client.configurePingHandling()
 
-	client.SendMessage(WEBSOCKET_AUTHENTICATION_CHALLENGE, map[string]interface{}{"token": authToken})
+	err = client.SendMessage(WEBSOCKET_AUTHENTICATION_CHALLENGE, map[string]interface{}{"token": authToken})
+	if err != nil {
+		return client, NewAppError("NewWebSocketClient", "model.websocket_client.authentication_challenge.app_error", nil, err.Error(), http.StatusInternalServerError)
```

Comment:
Should client not be returned here, since there was an error?

Reply:
Yes